### ISSUE TITLE
Start Java program from Eclipse causes ClassNotFoundExceptions

### ISSUE THREAD

**Issue by HannesWell** [oracle/graal#2774]
**Describe the issue**
Starting a Eclipse RCP or even a simple Java-Program from Eclipse causes multiple `ClassNotFoundException` to be thrown inside the "JVMCI-native CompilerThread0".

This issue is similar to issue #2710.

I also had the problem described #2710 when using "Eclipse Modelling Tools" to start an Eclipse-RCP.
With the new Graal CE 20.2 not only one ClassNotFoundException is thrown but multiple are thrown one after the other for the following classes (in the StackTrace at the end this is the first String argument named "cn"):
```
org.graalvm.compiler.hotspot.management.JMXToLibGraalCalls
org.graalvm.compiler.hotspot.management.JMXFromLibGraalEntryPoints
org.graalvm.compiler.hotspot.management.LibGraalMBean
org.graalvm.compiler.hotspot.management.LibGraalMBean$Factory
org.graalvm.compiler.hotspot.management.LibGraalMBean$PushBackIterator
org.graalvm.compiler.hotspot.management.AggregatedMemoryPoolBean
```
The second boolean argument named "resolve" of `loadClass()` is always `false`.

I also tested it with a freshly installed "Eclipse for Java Developers" and a simple Java-"Hello World" program. Even in this case the same ClassNotFoundExceptions are thrown!

**Steps to reproduce the issue**
1. Install Eclipse IDE for Java-Developers
2. Create a new Java Project containing a simple Java-_Hello-World_(or anything else) program with a main-method.
3. Right click, _Debug as Java application_ 
(make sure "Suspend execution on uncaught Exceptions" is enabled in the Eclipse preferences under _Java_, _Debugger_. It should be enabled by default.)

**Describe GraalVM and your environment:**
 - GraalVM version: CE 20.2.0
 - JDK major version: 11
 - OS: Debian GNU/Linux 10
 - Architecture: AMD64

**More details**
StackTrace of the uncaught Exception where the Eclipse debugger halts. It is equal for
```
Daemon System Thread [JVMCI-native CompilerThread0] (Suspended (exception ClassNotFoundException))	
	ClassLoaders$PlatformClassLoader(BuiltinClassLoader).loadClass(String, boolean) line: 581	
	ClassLoaders$PlatformClassLoader(ClassLoader).loadClass(String) line: 522	
```

**Comment by maxboehm** [comment]
Same here, it is a clean installation:
GraalVM: graalvm-ce-java11-20.2.0
Eclipse: eclipse-jee-2020-06-R-win32-x86_64
OS: Windows 10
Architecture: AMD64

**Comment by TodGilbertHarter** [comment]
I can confirm that with:
eclipse.buildId=4.17.0.I20200902-1800
java.runtime.name=OpenJDK Runtime Environment
java.runtime.version=11.0.8+10-jvmci-20.2-b03
java.specification.name=Java Platform API Specification
java.specification.vendor=Oracle Corporation
java.specification.version=11
java.vendor=GraalVM Community
java.vendor.url=https://www.graalvm.org/
java.vendor.url.bug=https://github.com/oracle/graal/issues
java.vendor.version=GraalVM CE 20.2.0
java.version=11.0.8
java.version.date=2020-07-14

This still happens. The only solution I found was to globally disable the debugger's "Suspend execution on uncaught exceptions" option. Kinda sub-optimal ;)

**Comment by tzezula** [comment]
The fix of the issue #2710 was integrated into the graalvm-20.3.

I've tried to reproduce this issue on the GraalVM CE 20.3.0  with java.specification.version=11 and no `ClassNotFoundException` were thrown. The fix of the issue #2710 resolved them.
On the JDK-11 there is still one `LinkageError` which I am fixing right  now.

Regarding the JDK-8 the JMX support needs to initialize JVMCI classloader which initializes the JVMCI runtime. This can cause some exceptions. I need to try. But the JVMCI classloader is no more used on the JDK-11.

**Comment by tzezula** [comment]
Works fine also on the `graalvm-ce-java8-20.3.0`.

**Comment by tzezula** [comment]
The `LinkageError` does not cause an application to stop on a breakpoint, it's harmless.
But it's still better to remove it.
This issue is fixed in the GraalVM-20.3.

**Comment by HannesWell** [comment]
With GraalVM CE 20.3 it works perfectly fine for me on Win10. Thanks for your support.

I'm just surprised that the fix of #2710 isn't integrated into Graal until 20.3. I thought the the fix would be contained in 20.2 already, because it was fixed before 20.2 was released and I didn't expect a delay. But probably it was too late to be considered.

Have you ever considered to (automatically) add tags like "fixed but not yet in master/ not yet released" and "fixed and integrated", so one can see if a fix is considered for the next release? I wouldn't have opened this issue if I had known that the fix was not yet integrated. So maybe this could save you same work.
